src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -899,53 +897,49 @@ private static void SetPathAndQueryParameters(CodeMethod target, OpenApiUrlTreeN
     private CodeTypeBase CreateModelDeclarations(OpenApiUrlTreeNode currentNode, OpenApiSchema schema, OpenApiOperation operation, CodeElement parentElement, string suffixForInlineSchema, OpenApiResponse response = default, string typeNameForInlineSchema = default)
     {
         var codeNamespace = parentElement.GetImmediateParentOfType<CodeNamespace>();
-        
+
         if (!schema.IsReferencedSchema() && schema.Properties.Any()) { // Inline schema, i.e. specific to the Operation
             return CreateModelDeclarationAndType(currentNode, schema, operation, codeNamespace, suffixForInlineSchema, typeNameForInlineSchema: typeNameForInlineSchema);
         } else if(schema.IsAllOf()) {
-            return CreateInheritedModelDeclaration(currentNode, schema, operation);
+            return CreateInheritedModelDeclaration(currentNode, schema, operation, codeNamespace);
         } else if((schema.IsAnyOf() || schema.IsOneOf()) && string.IsNullOrEmpty(schema.Format)) {
-            return CreateComposedModelDeclaration(currentNode, schema, operation, suffixForInlineSchema);
+            return CreateComposedModelDeclaration(currentNode, schema, operation, suffixForInlineSchema, codeNamespace);
```

Comment:
Pass over the `codeNamespace` to default to avoid regeneration/recalculation of the namespace. This also avoid having model types being duplicated/regenerated in other namespaces meant for request builders creating the possibility of ambiguous build errors if the two namespaces are both imported.